test/cuda/cufinufft2d1nupts_test.cu

```diff
@@ -190,53 +172,57 @@ int main(int argc, char *argv[]) {
     printf("[time  ] cufinufft exec (set 2):\t %.3g s\n", milliseconds / 1000);
 
     cudaEventRecord(start);
-    ier = CUFINUFFT_DESTROY(dplan);
+    ier = cufinufft_destroy_impl<T>(dplan);
     cudaEventRecord(stop);
     cudaEventSynchronize(stop);
     cudaEventElapsedTime(&milliseconds, start, stop);
     totaltime += milliseconds;
     printf("[time  ] cufinufft destroy:\t\t %.3g s\n", milliseconds / 1000);
 
-    checkCudaErrors(cudaMemcpy(fk1, d_fk1, N1 * N2 * sizeof(CUCPX), cudaMemcpyDeviceToHost));
-    checkCudaErrors(cudaMemcpy(fk2, d_fk2, N1 * N2 * sizeof(CUCPX), cudaMemcpyDeviceToHost));
+    fk1 = d_fk1;
+    fk2 = d_fk2;
 
     printf("[Method %d] (%d+%d) NU pts to %d U pts in %.3g s:      %.3g NU pts/s\n", opts.gpu_method, M1, M2, N1 * N2,
            totaltime / 1000, (M1 + M2) / totaltime * 1000);
     printf("\t\t\t\t\t(exec-only thoughput: %.3g NU pts/s)\n", (M1 + M2) / exec_ms * 1000);
 
     int nt1 = (int)(0.37 * N1), nt2 = (int)(0.26 * N2); // choose some mode index to check
-    CUFINUFFT_CPX Ft = CUFINUFFT_CPX(0, 0), J = IMA * (CUFINUFFT_FLT)iflag;
+    thrust::complex<T> Ft(0, 0), J(0, iflag);
     for (int j = 0; j < M1; ++j)
         Ft += c1[j] * exp(J * (nt1 * x1[j] + nt2 * y1[j])); // crude direct
     int it = N1 / 2 + nt1 + N1 * (N2 / 2 + nt2);            // index in complex F as 1d array
-    //	printf("[gpu   ] one mode: abs err in F[%ld,%ld] is %.3g\n",(int)nt1,(int)nt2,abs(Ft-fk[it]));
-    printf("[gpu   ] one mode: rel err in F[%ld,%ld] is %.3g (set 1)\n", (int)nt1, (int)nt2,
-           abs(Ft - fk1[it]) / infnorm(N, fk1));
-    Ft = CUFINUFFT_CPX(0, 0);
+
+    printf("[gpu   ] one mode: rel err in F[%d,%d] is %.3g (set 1)\n", (int)nt1, (int)nt2,
+           abs(Ft - fk1[it]) / infnorm(N, (std::complex<T> *)fk1.data()));
+    Ft = thrust::complex<T>(0, 0);
     for (int j = 0; j < M2; ++j)
         Ft += c2[j] * exp(J * (nt1 * x2[j] + nt2 * y2[j])); // crude direct
-    printf("[gpu   ] one mode: rel err in F[%ld,%ld] is %.3g (set 2)\n", (int)nt1, (int)nt2,
-           abs(Ft - fk2[it]) / infnorm(N, fk2));
-
-    cudaFreeHost(x1);
-    cudaFreeHost(y1);
-    cudaFreeHost(c1);
-    cudaFreeHost(fk1);
-    cudaFreeHost(x2);
-    cudaFreeHost(y2);
-    cudaFreeHost(c2);
-    cudaFreeHost(fk2);
-    cudaFree(d_x1);
-    cudaFree(d_y1);
-    cudaFree(d_c1);
-    cudaFree(d_fk1);
-    cudaFree(d_x2);
-    cudaFree(d_y2);
-    cudaFree(d_c2);
-    cudaFree(d_fk2);
-
-    // for cuda-memcheck to work
-    cudaDeviceReset();
+    printf("[gpu   ] one mode: rel err in F[%d,%d] is %.3g (set 2)\n", (int)nt1, (int)nt2,
+           abs(Ft - fk2[it]) / infnorm(N, (std::complex<T> *)fk2.data()));
 
     return 0;
```

Comment:
Yeah from the looks of it, the perf tester covers a lot of ground here, so it may be better to adapt that to do accuracy testing instead of relying on these older tests.